### ISSUE TITLE
Improved handling of content strings / errors in the Recipient Tool

### ISSUE THREAD

**Issue by jduke99** [langroid/langroid#597]
**Is your feature request related to a problem? Please describe.**
Recipient tool is of course critical to the utility of Langroid. While ChatGPT produces picture perfect handling of the recipient tool, other LLMs such as Llama 3.1 are just a bit off. In particular, they can be inclined to put the content in as JSON rather than as a string.  Langroid handles this reasonably gracefully, reminding the LLM to please reformat the request.  But this process is time consuming and doesn't always work as desired.  

**Describe the solution you'd like**
A first pass look at the LLM result which determines that the recipient tool is being called could also modify the content before passing on to the Recipient Tool class itself. Specifically, can you check for non-quoted content and automatically add quotes, change double quotes to single quotes inside or do the necessary escaping inside so that it can be handled by the Recipient Tool without error. 

**Describe alternatives you've considered**
I have tried to change my prompts in many ways as well as the prompt message inside the Recipient Tool to no success.

**Additional context**
This small change would be immensely helpful in enabling production use where we cannot use commercial LLMs and are limited to slightly weaker models.

**Comment by pchalasani** [comment]
@jduke99 thanks for this suggestion. Please if this FAQ (and related example) I just added helps -- 
https://langroid.github.io/langroid/FAQ/#how-can-i-deal-with-llms-especially-weak-ones-generating-bad-json-in-tools

TLDR - weak LLMs often return bad/hard-to-fix JSON in tools. XML-based tools can be much more reliable. Also `SendTool` is simpler to use than `RecipientTool`.

**Comment by pchalasani** [comment]
closing/stale
